samples/general/mobile_nerf_rayquery/mobile_nerf_rayquery.cpp

```diff
@@ -0,0 +1,1252 @@
+
+	// Required for ray queries
+	add_device_extension(VK_KHR_RAY_QUERY_EXTENSION_NAME);
+
+	// Required by VK_KHR_spirv_1_4
+	add_device_extension(VK_KHR_SPIRV_1_4_EXTENSION_NAME);
+	add_device_extension(VK_KHR_SHADER_FLOAT_CONTROLS_EXTENSION_NAME);
+
+	// Use this extension for better storage buffers layout
+	add_device_extension(VK_EXT_SCALAR_BLOCK_LAYOUT_EXTENSION_NAME);
+}
+
+MobileNerfRayQuery::~MobileNerfRayQuery()
+{
+	if (has_device())
+	{
+		auto device_ptr = get_device().get_handle();
+
+		for (auto &model : models)
+		{
+			model.vertex_buffer.reset();
+			model.index_buffer.reset();
+
+			vkDestroySampler(get_device().get_handle(), model.texture_input_0.sampler, nullptr);
+			vkDestroySampler(get_device().get_handle(), model.texture_input_1.sampler, nullptr);
+		}
+
+		vkDestroyPipeline(device_ptr, pipeline, nullptr);
+		vkDestroyPipelineLayout(device_ptr, pipeline_layout, nullptr);
+		vkDestroyDescriptorSetLayout(device_ptr, descriptor_set_layout_common, nullptr);
+		vkDestroyDescriptorSetLayout(device_ptr, descriptor_set_layout_indices, nullptr);
+		vkDestroyDescriptorSetLayout(device_ptr, descriptor_set_layout_vertices, nullptr);
+		vkDestroyDescriptorSetLayout(device_ptr, descriptor_set_layout_feature1, nullptr);
+		vkDestroyDescriptorSetLayout(device_ptr, descriptor_set_layout_feature2, nullptr);
+
+		for (auto &weights_buffer : weights_buffers)
+			weights_buffer.reset();
+
+		uniform_buffer.reset();
+	}
+}
+
+bool MobileNerfRayQuery::prepare(const vkb::ApplicationOptions &options)
+{
+	read_json_map();
+
+	// Load the mlp for each model
+	mlp_weight_vector.resize(num_models);
+	for (int i = 0; i < num_models; i++)
+	{
+		initialize_mlp_uniform_buffers(i);
+	}
+
+	if (!ApiVulkanSample::prepare(options))
+	{
+		return false;
+	}
+
+	load_shaders();
+
+	// Setup camera
+	camera.type  = vkb::CameraType::LookAt;
+	camera_pos.y = -camera_pos.y;
+	camera_set_look_at(camera, camera_pos, glm::vec3(0.0f), glm::vec3(0.0f, 1.0f, 0.0f));
+
+	camera.set_perspective(60.0f, static_cast<float>(width) / static_cast<float>(height), 0.01f, 256.0f);
+
+	// Each models may have submodels
+	int models_entry = 0;
+	for (int model_index = 0; model_index < num_models; model_index++)
+	{
+		int num_sub_model = models[models_entry].sub_model_num;
+
+		for (int sub_model_index = 0; sub_model_index < num_sub_model; sub_model_index++)
+		{
+			load_scene(model_index, sub_model_index, models_entry);
+			create_texture(model_index, sub_model_index, models_entry);
+			create_static_object_buffers(models_entry);
+			create_bottom_level_acceleration_structure(models_entry);
+			models_entry++;
+		}
+	}
+
+	create_top_level_acceleration_structure();
+	create_uniforms();
+	create_pipeline_layout();
+	create_descriptor_pool();
+	create_descriptor_sets();
+	prepare_pipelines();
+	build_command_buffers();
+
+	prepared = true;
+	LOGI("Prepare Done!");
+	return true;
+}
+
+void MobileNerfRayQuery::request_gpu_features(vkb::PhysicalDevice &gpu)
```

Comment:
With #1013 now merged, this needs to be adjusted accordingly. Probably best to update this to master and then update this part of the code.